sql/_05_plcsql/_01_testspec/_04_expression/_07_sql_rowcount/_01_basic/cases/bug_after_commit_rollback.sql

```diff
@@ -0,0 +1,31 @@
+--+ server-message on
+-- Vefified for CBRD-24962
+-- SQL%ROWCOUNT must be zero after COMMIT AND ROLLBACK
+
+create or replace procedure t as
+	i int;
+begin
+
+	select charset_id into i from db_charset where charset_name='utf8';
+	dbms_output.put_line(sql%rowcount);
+	rollback;
+	dbms_output.put_line(sql%rowcount);
+end;
+
+call t();
+
+create or replace procedure t as
+	i int;
+begin
+
+	select charset_id into i from db_charset where charset_name='utf8';
+	dbms_output.put_line(sql%rowcount);
+	commit;
+	dbms_output.put_line(sql%rowcount);
+end;
+
+call t();
+
```

Comment:
create table test ( col1 int  );
insert into test values(1),(2),(3),(4),(5);

create or replace procedure t as
	i int;
begin
	insert into test values(1),(2),(3),(4),(5);
	dbms_output.put_line(sql%rowcount);
	commit;
	dbms_output.put_line(sql%rowcount);
	
	update test set col1 = col1 +1 ;
	dbms_output.put_line(sql%rowcount);
	commit;
	dbms_output.put_line(sql%rowcount);
	
	delete from test where col1 < 5 ;
	dbms_output.put_line(sql%rowcount);
	commit;
	dbms_output.put_line(sql%rowcount);
	
end;

call t();


drop table test ;
create table test ( col1 int  );
insert into test values(1),(2),(3),(4),(5);

create or replace procedure t as
	i int;
begin
	insert into test values(1),(2),(3),(4),(5);
	dbms_output.put_line(sql%rowcount);
	rollback;
	dbms_output.put_line(sql%rowcount);
	
	update test set col1 = col1 +1 ;
	dbms_output.put_line(sql%rowcount);
	rollback;
	dbms_output.put_line(sql%rowcount);
	
	delete from test where col1 < 5 ;
	dbms_output.put_line(sql%rowcount);
	rollback;
	dbms_output.put_line(sql%rowcount);
	
end;

call t();

Reply:
Thx, I reflected that